tests/tensor/test_einsum.py

```diff
@@ -281,3 +282,15 @@ def test_threeway_mul(static_length):
         out.eval({x: x_test, y: y_test, z: z_test}),
         np.full((3,), fill_value=6),
     )
+
+
+def test_repeated_inputs():
+    x = matrix("x")
+    out_repeated = einsum("ij,ij->i", x, x)
+    out_copy = einsum("ij,ij->i", x, x.copy())
+
+    x_test = np.array([[1, 2], [3, 4]])
```

Comment:
You need this, for the one test that's failing
```suggestion
    x_test = np.array([[1, 2], [3, 4]]).astype(x.dtype)
```